test/from-draft3/to-draft4/required-is-absent-and-required-true-in-properties.json

```diff
@@ -0,0 +1,16 @@
+[
+    {
+        "title": "`required` is absent in outer schema and there is some properties/{something}/required true",
+        "from": {
+            "properties": {
+                "foo": { "required": true }
+            }
+        },
+        "to": {
+            "properties": {
+                "foo": { "required": true }
```

Comment:
Shouldn't this disappear and `foo` appear on the array below?

Reply:
Huh.. no its a pre-processing step to initiate required with empty array, 
Reason: If we try to have creation and remove-append in a single rule then creation of empty array would happen every time...

Reply:
Right, but we should not be testing preprocessing steps here. Remember this test suite is meant to test the system as a whole, not individual rules. i.e. what is the result of transforming the input schema after all the rules that could have possible applied have applied?

Reply:
@jviotti  so for each test we will be applying all the rules together?

Reply:
Correct. We are never testing rules on their own